src/System.Management.Automation/engine/CommandInfo.cs

```diff
@@ -529,7 +529,7 @@ private void GetMergedCommandParameterMetadata(out MergedCommandParameterMetadat
                 processor = scriptCommand != null
                     ? new CommandProcessor(scriptCommand, _context, useLocalScope: true, fromScriptFile: false,
                         sessionState: scriptCommand.ScriptBlock.SessionStateInternal ?? Context.EngineSessionState)
-                    : new CommandProcessor((CmdletInfo)this, _context) { UseLocalScope = true };
```

Comment:
This is because `UseLocalScope` is only used for script commands (namely only when `scriptCommand != null`). It's not used at all when `CommandProcessor` is constructed with `CmdletInfo`. I spent some time trying to understand why `{ UseLocalScope = true }` is used here, and think it's probably better to remove it since `UseLocalScope` is meaningless for `CmdletInfo`, so that we don't spend more time in future trying to understand this code again.

Reply:
Maybe update the comment in UseLocalScope definition?
```
        /// <summary>
        /// For commands that use the scope stack, if this flag is
        /// true, don't create a new scope when running this command.
        /// </summary>
```